### PR TITLE
Lora loading for bf16 and fp8, as separate models

### PR DESCRIPTION
**Pull request by andreasjansson**
Had to fix some bugs in the original lora loading code.

Outputs are here: https://replicate.com/replicate-internal/test-flux-dev-lora
* bf16 lora loading works
* For fp8 lora loading, the lora strength is much lower and I need to crank up lora_scale to 1.7 to match lora_scale 1.1 at bf16

Fusing and unfusing is also slightly lossy so the model slowly degrades over time. We could do something like what peft and add a new node that does the matmul on the fly, instead of fusing. But that would slow down inference. Curious if you have ideas @daanelson

### CODE REVIEW

**Comment by Averylamp** [comment]
Very excited for this PR.  Thanks for doing this!  I was looking for an H100 lora inference provider and this seems like it would do the trick.  I was curious as well if pricing for fast generations would be any different than per image because the GPU usage time is much less?

**Comment by Averylamp** [comment]
Hi, I was curious if this work is continuing on this PR?  I believe this should make flux dev lora inference fast enough that you'd gain a customer versus using Fal as they currently are a few seconds faster (but from my benchmarking, this should be faster in the end).  Happy to take on any tasks if you would like as well.

File: .github/workflows/push.yaml [hunk 1]
```diff
@@ -55,3 +56,50 @@ jobs:
         REPLICATE_API_TOKEN: ${{ secrets.REPLICATE_API_TOKEN }}
       run: |
         cog-safe-push -vv ${{ github.event.inputs.no_push == 'true' && '--no-push' || '' }} --config=cog-safe-push-dev.yaml
+
```
Comment (daanelson) [inline .github/workflows/push.yaml]: whoops

File: cog-safe-push-dev-lora.yaml [hunk 2]
```diff
@@ -0,0 +1,103 @@
+model: replicate/flux-dev-lora-internal-model
```
Comment (daanelson) [inline cog-safe-push-dev-lora.yaml]: have enough of these that it makes sense to move into a subdirectory

File: fp8/lora_loading.py [hunk 3]
```diff
@@ -365,32 +487,38 @@ def apply_lora_weight_to_module(
...
         w_up = w_up * alpha / rank
     if uneven_rank:
-        fused_lora = lora_scale * torch.mm(
-            w_down.repeat_interleave(int(rank_diff), dim=1), w_up
-        )
-    else:
-        fused_lora = lora_scale * torch.mm(w_down, w_up)
-    fused_weight = w_orig + fused_lora
-    return fused_weight.to(dtype=w_dtype, device=device)
+        w_down = w_down.repeat_interleave(int(rank_diff), dim=1)
```
Comment (daanelson) [inline fp8/lora_loading.py:486]: clean up

**Review by daanelson — APPROVED** [review]
clean up a few things and then we're good!

**Pull request merged**